### PR TITLE
[SYS-678]: Allow user to provide connection parameters

### PR DESCRIPTION
**Pull request by akatch**
Users may now provide `--scheme`, `--host`, `--port`, `--username`, `--password`, and `--vhost` parameters at the command line.

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -1,93 +1,113 @@
...
+func (suite *MQPTestSuite) TestNewBasicPublishing() {
 	/*
 	   cases:
 	   - basic alphanumeric/special chars string
 	   - unicode string
 	   - emoji string
 	*/
 	pub := newBasicPublishing("hello world")
-	assert.Equal(t, string(pub.Body), "hello world")
+	assert.Equal(suite.T(), string(pub.Body), "hello world")
```
Comment (douglb) [inline main_test.go]: Minor item.  The `assert.Equal` call takes (t, expected, actual) and it looks like you are passing (t, actual, expected) .  The equality check will still work fine and fail correctly when they don't match.  However, the error message the function produces on error will make it look like your actual value was the expected one which is misleading.

I'm also assuming the cases comment is a todo list of checks you want to implement?

File: main_test.go [hunk 2]
```diff
@@ -1,93 +1,113 @@
...
-func TestPublishMessage(t *testing.T) {
+func (suite *MQPTestSuite) TestPublishMessage() {
 	/*
 	   cases:
 	   - basic known good single message
 	   - multiple messages
 	*/
-	conn := setup()
+	//conn := setup()
+	conn := suite.mq.Connection
```
Comment (douglb) [inline main_test.go]: If you're no opening the connection, this test should probably not be closing the connection at the end of it.

File: main_test.go [hunk 3]
```diff
@@ -1,93 +1,113 @@
...
+type MQPTestSuite struct {
+	suite.Suite
+	mq *MQ
 }
 
-func TestConnect(t *testing.T) {
+func (suite *MQPTestSuite) SetupSuite() {
+	uri := parseFlags()
+	var err error
+	suite.mq, err = NewMQ(uri)
```
Comment (douglb) [inline main_test.go]: The `NewMQ` call will do the actual `connect()` to the server.  I don't see a `TeadDownSuite` to do the close of the connection?

File: main_test.go (hunk 1 above)
Comment (akatch) [reply main_test.go]: > I'm also assuming the cases comment is a todo list of checks you want to implement?

Correct. I went through and corrected all the backwards `assert.Equal` calls - thank you for catching this.

**Comment by akatch** [comment]
I believe I've addressed all of the feedback to this point, please call it out if you see something I missed.

**Pull request merged**